code/__HELPERS/lists.dm

```diff
@@ -344,3 +344,13 @@
 	while(L.Remove(null))
 		continue
 	return L
+
+//Copies a list, and all lists inside it recusively
+//Does not copy any other reference type
+/proc/deepCopyList(list/l)
+	if(!islist(l))
+		return l
+	. = l.Copy()
+	for(var/i = 1 to length(.))

```

Comment:
why not ```l.len```? ```.``` isn't pathed as a list so you obviously can't do it there but it's a perfect copy of ```l```